### ISSUE TITLE
Output only Comes at Irregular Intervals when it's not Directly to Stdout

### ISSUE THREAD

**Issue by DeedleFake** [fireice-uk/xmr-stak-cpu#342]
I'm running the miner as a systemd unit. Output from the process is being redirected to the journal, as is the default, but it doesn't seem to be writing to stdout when it should be. If I just run `xmr-stak-cpu` directly, it writes to stdout every time, for example, a result is accepted, which is fairly constant, but if I pipe the output into `less` or if I run it via systemd, it only seems to write output every minute or so.

I have `daemon_mode` enabled in `config.txt`.

**Comment by psychocrypt** [comment]
Could it be that io of background processes are buffered?


Am 20.09.2017 15:49 schrieb "DeedleFake" <notifications@github.com>:

**Comment by DeedleFake** [comment]
I don't think so. It only seems to happens with xmr-stak-cpu. Just to test, I threw the following unit and script together and it worked fine:

test.service
```systemd
[Unit]
Description=Test

[Service]
ExecStart=/tmp/test.sh
```

test.sh
```bash
#!/usr/bin/bash

for ((i = 0; i < 100; i++)); do
	echo "Test: $i"
	sleep .5s
done
```

I ran `journalctl -af --user-unit test.service` and it printed a line every half a second.

**Comment by psychocrypt** [comment]
have you enabled the deamon mode in the config?

Am 25.09.2017 3:04 Nachm. schrieb "DeedleFake" <notifications@github.com>:

**Comment by DeedleFake** [comment]
Yep.

**Comment by darugnaa** [comment]
I had the same problem. this behaviour depends on whether STDOUT is a pipe or a shell.
If STDOUT is a shell, it's unbuffered by default. (write each line immediately)
If STDOUT is a pipe, it's buffered by default. (write the lines when they fill the buffer, which may be a lot later)

opened a pr https://github.com/fireice-uk/xmr-stak-cpu/pull/352 with an option to change this behaviour if desired.

Added the option to forcefully flush STDOUT, which may be useful when running the miner through a pipe instead than a shell.

**Comment by darugnaa** [comment]
@DeedleFake  compile, add `"flush_stdout": true,` option (see the change in config.txt) and it should work as expected
